tests/model/test_core.py

```diff
@@ -527,6 +527,42 @@ def test_model_var_maps():
     assert model.rvs_to_transforms[x] is None
 
 
+class TestTransformArgs:
+    def test_transform_warning(self):
+        with pm.Model():
+            with pytest.warns(
+                UserWarning,
+                match="To disable default transform,"
+                " please use default_transform=None"
+                " instead of transform=None. Setting transform to"
+                " None will not have any effect in future.",
+            ):
+                a = pm.Normal("a", transform=None)
+
+    def test_transform_order(self):
+        transform_order = []
+
+        class DummyTransform(Transform):
+            name = "dummy1"
+            ndim_supp = 0
+
+            def __init__(self, marker) -> None:
+                super().__init__()
+                self.marker = marker
+
+            def forward(self, value, *inputs):
+                nonlocal transform_order
+                transform_order.append(self.marker)
+                return value
+
+            def backward(self, value, *inputs):
+                return value
+
+        with pm.Model() as model:
+            x = pm.Normal("x", transform=DummyTransform(2), default_transform=DummyTransform(1))
+        assert transform_order == [1, 2]
```

Comment:
You can use regular transforms, and simply assert the obtained transform is a Chain, which as a property like `transform_list` that includes the transforms, and you can assert those are also the expected ones. The transform is available in `models.rvs_to_transforms[x]`

Also, would be nice to include a numerical example that would have led to `nan` or `-inf` probability before the change, like an `ordered mixture of LogNormals` evaluated at -1